### PR TITLE
remove traceur from the build pipeline

### PR DESCRIPTION
**Pull request by rkirov**
npm traceur package is still used for traceur-runtime dependency.

### CODE REVIEW

**Comment by rkirov** [comment]
@pkozlowski-opensource The typescript version we are using produces bad -m system output. When trying to upgrade to a newer one I run into slew of other errors. I am sure that we can get it to work (maybe with some help from TS and system.js guys). However, since removing traceur has been pending for quite long on our task list, I rather go ahead with this option, because it just works.

 Note that system.js is still used as a module loader for the commonjs output for the e2e and unit tests runs, which means

File: tools/broccoli/trees/browser_tree.ts [hunk 1]
```diff
@@ -109,11 +115,19 @@ module.exports = function makeBrowserTree(options, destinationPath) {
...
+    mapRoot: '',  // force sourcemaps to use relative path
+    module: 'commonjs',
+    noEmitOnError: false,
+    rootDir: '.',
+    sourceMap: true,
+    sourceRoot: '.',
+    target: 'ES5'
   });
 
   // Now we add a few more files to the es6 tree that Traceur should not see
```
Comment (alexeagle) [inline tools/broccoli/trees/browser_tree.ts]: update comment

**Comment by alexeagle** [comment]
Fix the failing tests.
Seems like a great next step to me. Do we have any bugs open for the problems with TS System emit?

**Comment by rkirov** [comment]
@alexeagle the issue is the one you know about https://github.com/systemjs/systemjs/issues/712 and from some local testing with typescript@next it looks like it is fixed at TS head. However, upgrading TS leads to other jankiness. Lets get this in, upgrade TS and then try to flip es5 output back to 'system' (if we are still convinced that is important).

**Comment by rkirov** [comment]
PTAL

File: modules/examples/e2e_test/sourcemap/sourcemap_spec.ts [hunk 2]
```diff
@@ -34,15 +34,10 @@ describe('sourcemaps', function() {
...
-      var finalMapData = fs.readFileSync('dist/js/prod/es6/examples/src/sourcemap/index.js.map');
-      var finalDecoder = new sourceMap.SourceMapConsumer(JSON.parse(finalMapData));
-
-      var finalPosition = finalDecoder.originalPositionFor(originalPosition);
-
       var sourceCodeLines =
           fs.readFileSync('modules/examples/src/sourcemap/index.ts', {encoding: 'UTF-8'})
               .split('\n');
-      expect(sourceCodeLines[finalPosition.line - 1])
+      expect(sourceCodeLines[originalPosition.line - 1])
```
Comment (IgorMinar) [inline modules/examples/e2e_test/sourcemap/sourcemap_spec.ts]: doesn't this test now test that we don't have source mapping at all?

File: tools/broccoli/trees/browser_tree.ts (hunk 1 above)
Comment (IgorMinar) [inline tools/broccoli/trees/browser_tree.ts]: +1

File: tools/broccoli/trees/node_tree.ts [hunk 3]
```diff
@@ -34,7 +34,7 @@ module.exports = function makeNodeTree(destinationPath) {
     experimentalDecorators: true,
     declaration: false,
     mapRoot: '', /* force sourcemaps to use relative path */
-    module: 'commonjs',
+    module: 'CommonJS',
```
Comment (IgorMinar) [inline tools/broccoli/trees/node_tree.ts]: is this change needed?

Comment (rkirov) [inline tools/broccoli/trees/node_tree.ts]: nope, picked it up from Ian. I believe TS normalizes the string. It would complain if it receives a module format it doesn't recognize. Will revert.

File: modules/examples/e2e_test/sourcemap/sourcemap_spec.ts (hunk 2 above)
Comment (rkirov) [inline modules/examples/e2e_test/sourcemap/sourcemap_spec.ts]: Nope, before we did ts -> es6 -> es5 in two shots, and didn't properly merge the sourcemaps for the two jumps. So in the test we had to decode twice (once per transpilation).

Now, we have a direct ts -> es5, so we just need to read in one source map and compare with the original. Notice we read the expected line from modules/examples/src/sourcemap/index.ts

**Comment by alexeagle** [comment]
Note that we already cherry-picked one upstream TS-system fix into the TS we use for Angular
https://github.com/alexeagle/TypeScript/commit/6470f8e3a08f5d27b77f0ca3fb7532f97fb1fc70
so we can easily do that again if we are missing another bugfix.

**Comment by rkirov** [comment]
@alexeagle right, but I rather put effort into bringing newer versions that are released on npm (currently looks like 1.6.0-beta), and avoid too much cherry-picking. Also system vs commonjs module format output doesn't seem critical to me, since system.js loads them both fine.

File: tools/broccoli/trees/browser_tree.ts (hunk 1 above)
Comment (rkirov) [inline tools/broccoli/trees/browser_tree.ts]: done.

File: tools/broccoli/trees/node_tree.ts (hunk 3 above)
Comment (rkirov) [inline tools/broccoli/trees/node_tree.ts]: Nope, I was wrong, since we don't go through the CLI interface we have to pass a string that is a key in the following const enum https://github.com/Microsoft/TypeScript/blob/master/lib/typescript.d.ts#L1344
